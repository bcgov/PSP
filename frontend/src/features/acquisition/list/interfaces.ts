export interface IAcquisitionFilter {
  acquisitionFileStatusTypeCode: string;
  acquisitionFileNameOrNumber: string;
  projectNameOrNumber: string;
}
