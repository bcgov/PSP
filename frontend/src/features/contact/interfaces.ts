export interface IContactFilter {
  summary: string;
  municipality: string;
  activeContactsOnly: boolean;
  searchBy: string;
}
