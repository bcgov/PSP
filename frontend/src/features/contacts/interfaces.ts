export interface IContactFilter {
  summary: string;
  municipality: string;
  activeContactsOnly: boolean;
  searchBy: string;
}

export enum ContactTypes {
  ORGANIZATION = 'O',
  INDIVIDUAL = 'P',
}
