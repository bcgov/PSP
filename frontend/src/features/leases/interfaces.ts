export interface ILeaseFilter {
  pinOrPid: string;
  lFileNo: string;
  searchBy: string;
  tenantName: string;
  programs: string[];
}
