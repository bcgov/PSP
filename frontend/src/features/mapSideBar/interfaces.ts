import { AddressTypes, PropertyTypes } from 'constants/index';
import { IAddress } from 'interfaces';

export const defaultPropertyValues = {
  address: {
    addressTypeId: AddressTypes.Physical,
    streetAddress1: '',
    municipality: '',
    postal: '',
    province: '',
    provinceId: 1,
    region: '',
    district: '',
  } as IAddress,
  pid: '',
  propertyTypeId: PropertyTypes.Land,
  classification: '',
  region: '',
  district: '',
  zoning: '',
  landArea: '',
  landLegalDescription: '',

  // TODO: Theses properties probably should be replaced or updated.
  titleNumber: '',
  purpose: '',
  ruralArea: '',
  ocpDesignation: '',
};
