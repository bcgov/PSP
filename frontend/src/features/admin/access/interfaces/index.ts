import { AccessRequestStatus } from 'constants/accessStatus';

export interface IAccessRequestModel {
  id: number;
  userId: number;
  businessIdentifierValue: string;
  firstName: string;
  surname: string;
  email: string;
  position: string;
  status: AccessRequestStatus;
  organization: string;
  role: string;
  note: string;
}
