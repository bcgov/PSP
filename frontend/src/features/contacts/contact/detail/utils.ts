import { IContactAddress } from 'interfaces/IContact';

import { AddressTypes } from './../../../../constants/addressTypes';

// the order of this array corresponds to the expected display order
const addressSortOrder = [AddressTypes.Mailing, AddressTypes.Residential, AddressTypes.Billing];

export const sortAddresses = (a1: IContactAddress, a2: IContactAddress) => {
  if (a2.addressType.id === a1.addressType.id) {
    return 0;
  }
  const a2Index = addressSortOrder.indexOf(a2.addressType.id as AddressTypes);
  if (a2Index !== -1 && a2Index < addressSortOrder.indexOf(a1.addressType.id as AddressTypes)) {
    return 1;
  }

  return -1;
};

export const fakeAddresses = [
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'BILLING',
      description: 'Billing address',
      isDisabled: false,
    },
    streetAddress1: 'Billing address',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6A 5G7',
  },
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'MAILING',
      description: 'Mailing address',
      isDisabled: false,
    },
    streetAddress1: 'Mailing address',
    streetAddress2: 'Living in a van',
    streetAddress3: 'Down by the River',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6Z 5G7',
  },
  {
    id: 5,
    rowVersion: 1,
    addressType: {
      id: 'RESIDNT',
      description: 'Property address',
      isDisabled: false,
    },
    streetAddress1: 'Property address',
    streetAddress2: '',
    streetAddress3: '',
    municipality: 'vic',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V9A 0H6',
  },
];
