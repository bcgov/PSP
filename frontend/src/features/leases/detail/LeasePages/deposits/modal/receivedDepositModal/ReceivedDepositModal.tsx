import * as CommonStyled from 'components/common/styles';
import { FormikProps } from 'formik';
import { FormLeaseDeposit } from 'interfaces';
import * as React from 'react';
import { useRef } from 'react';

import { ReceivedDepositForm } from './ReceivedDepositForm';

export interface IReceivedDepositModalProps {
  initialValues: FormLeaseDeposit;
  display: boolean;
  onCancel: () => void;
  onSave: (values: FormLeaseDeposit) => void;
}

/**
 * Modal displaying form allowing add/update lease deposits. Save button triggers internal formik validation and submit.
 * @param {IReceivedDepositModalProps} props
 */
export const ReceivedDepositModal: React.FunctionComponent<IReceivedDepositModalProps> = ({
  initialValues,
  display,
  onCancel,
  onSave,
}) => {
  const formikRef = useRef<FormikProps<FormLeaseDeposit>>(null);
  const modalTitle = initialValues.id === undefined ? 'Add a Deposit' : 'Edit Deposit';
  return (
    <CommonStyled.PrimaryGenericModal
      title={modalTitle}
      display={display}
      okButtonText="Save"
      cancelButtonText="Cancel"
      handleCancel={onCancel}
      handleOk={() => {
        formikRef?.current?.submitForm();
      }}
      message={
        <ReceivedDepositForm formikRef={formikRef} initialValues={initialValues} onSave={onSave} />
      }
    />
  );
};

export default ReceivedDepositModal;
