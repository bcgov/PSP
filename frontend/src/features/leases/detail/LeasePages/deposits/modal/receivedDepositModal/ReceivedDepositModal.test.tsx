import userEvent from '@testing-library/user-event';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { createMemoryHistory } from 'history';
import { FormLeaseDeposit } from 'interfaces';
import { mockLookups } from 'mocks/mockLookups';
import { lookupCodesSlice } from 'store/slices/lookupCodes';
import { fillInput, renderAsync, RenderOptions, waitFor } from 'utils/test-utils';

import ReceivedDepositModal, { IReceivedDepositModalProps } from './ReceivedDepositModal';

const history = createMemoryHistory();
const mockAxios = new MockAdapter(axios);
const onSave = jest.fn();
const onCancel = jest.fn();
const storeState = {
  [lookupCodesSlice.name]: { lookupCodes: mockLookups },
};

describe('ReceivedDepositModal component', () => {
  const setup = async (
    renderOptions: RenderOptions &
      Partial<IReceivedDepositModalProps> & {
        initialValues?: any;
      } = {},
  ) => {
    // render component under test
    const component = await renderAsync(
      <ReceivedDepositModal
        onSave={onSave}
        onCancel={onCancel}
        display={true}
        initialValues={FormLeaseDeposit.createEmpty()}
      />,
      {
        ...renderOptions,
        history,
        store: storeState,
      },
    );

    return {
      component,
    };
  };

  beforeEach(() => {
    mockAxios.resetHistory();
    onSave.mockReset();
    onCancel.mockReset();
  });
  it('renders as expected', async () => {
    const { component } = await setup({});

    expect(component.asFragment()).toMatchSnapshot();
  });
  it('renders with data as expected', async () => {
    const { component } = await setup({
      initialValues: FormLeaseDeposit.createEmpty(),
    });

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('submits all filled out fields as expected', async () => {
    const {
      component: { getByText },
    } = await setup({});

    await fillInput(document.body, 'depositTypeCode', 'SECURITY', 'select');
    await fillInput(document.body, 'description', 'Test description', 'textarea');
    await fillInput(document.body, 'amountPaid', '1235');
    await fillInput(document.body, 'depositDate', '2020-01-02', 'datepicker');
    const saveButton = getByText('Save');
    userEvent.click(saveButton);
    await waitFor(() => expect(onSave).toHaveBeenCalled());
    expect(onSave).toHaveBeenCalledWith({
      amountPaid: 1235,
      depositDate: '2020-01-02',
      depositTypeCode: 'SECURITY',
      description: 'Test description',
      id: undefined,
      organizationDepositHolderId: '',
      otherTypeDescription: '',
      personDepositHolderId: '',
      rowVersion: 0,
    });
  });

  it('calls onCancel when cancel button clicked', async () => {
    const {
      component: { getByText },
    } = await setup({});
    const cancelButton = getByText('Cancel');
    userEvent.click(cancelButton);
    expect(onCancel).toHaveBeenCalled();
  });
});
