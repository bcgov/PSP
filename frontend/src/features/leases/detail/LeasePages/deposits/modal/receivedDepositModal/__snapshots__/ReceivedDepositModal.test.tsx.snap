// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceivedDepositModal component renders as expected 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div
    class="Toastify"
  />
  <div
    class="container"
  />
</DocumentFragment>
`;

exports[`ReceivedDepositModal component renders with data as expected 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div
    class="Toastify"
  />
  <div
    class="container"
  />
</DocumentFragment>
`;
